src/components/CommonComponents/Pagination/utils/buildComponentData.ts

```diff
@@ -0,0 +1,92 @@
+import { createElement, MouseEvent, ReactElement } from 'react';
+import { LocalizedLink as Link } from 'gatsby-theme-i18n';
+import { FormattedMessage } from 'react-intl';
+import { PageType } from './buildPaginationModel';
+
+type OnClick = {
+  (event: MouseEvent): void;
+};
+
+interface Props extends Partial<HTMLElement> {
+  as: string;
+  onClick?: OnClick;
+  to?: string;
+}
+
+interface ComponentData {
+  props: Props;
+  key: string;
+  content: string | ReactElement;
+}
+
+export function buildComponentData(
```

Comment:
I'm not sure about the naming of this utility. It doesn't say which component it is building data for.

Reply:
fixed